In the era of growing bioinformatic databases in the public domain, querying them programmatically becomes an essential part of any data or knowlege pipeline. However, pratical information of using the API of public databases are lacking. Here I present some practical scripts for accessing the NCBI entrez databases.
Entrez is an umbrella of databases, such as Nucleotide, Protein, Structure, Domains, OMIM, PubmMed, SNP, and Taxonomy. eUtils are set of seven programs that provide an interface to entrez query and database system at NCBI.The two technologies for accessing entrez/eUtils are the Web Common Gateway Interface (CGI), and Remote Procedure Calls(RPC) over SOAP (simple object access protocol), known as JAX-RPC or "Web Services".
CGI scripts are in perl, implementing the efetch, esearch, elink routines, and these scripts were tested to work as expected. A sample SOAP web services program are also presented, retrieving, for example, PubMed article abstracts, and it was coded in Java, and I did not test it extensively(just for a comparison with CGI).
